HiP-DataStore/Controllers/MediaController.cs

```diff
@@ -78,16 +79,22 @@ public IActionResult Get(MediaQueryArgs args)
                     .FilterByStatus(args.Status)
                     .FilterIf(args.Used != null, x => x.IsUsed == args.Used)
                     .FilterIf(args.Type != null, x => x.Type == args.Type)
-                    .FilterIf(args.Timestamp != null, x => DateTimeOffset.Compare(x.Timestamp, args.Timestamp.GetValueOrDefault()) == 1)
-                    .FilterIf(!string.IsNullOrEmpty(args.Query), x =>
+                     .FilterIf(!string.IsNullOrEmpty(args.Query), x =>
                         x.Title.ToLower().Contains(args.Query.ToLower()) ||
                         x.Description.ToLower().Contains(args.Query.ToLower()))
                     .Sort(args.OrderBy,
                         ("id", x => x.Id),
                         ("title", x => x.Title),
                         ("timestamp", x => x.Timestamp))
                     .Paginate(args.Page, args.PageSize)
-                    .Select(x => new MediaResult
+                    .ToList();
+
+                medias = medias.AsQueryable().FilterIf(args.Timestamp != null, x => DateTimeOffset.Compare(x.Timestamp, args.Timestamp.GetValueOrDefault()) == 1).ToList();
```

Comment:
I guess your problem was that MongoDB does not support DateTimeOffset.Compare(...) in queries, right? Hm, I'll have a look at that issue and maybe find a workaround.